plugin_tests/submission_test.py

```diff
@@ -0,0 +1,395 @@
+            }
+        )
+        self.assertStatusOk(resp)
+        self.assertDictContains({
+            'creatorId': str(self.user['_id']),
+            'creatorName': 'First Last',
+            'folderId': str(folder['_id']),
+            'phaseId': str(self.phase1['_id']),
+            'title': 'submission phase 1'
+        }, resp.json)
+        id = resp.json['_id']
+
+        # user cannot modify submissions
+        resp = self.request(
+            path='/covalic_submission/%s' % id, method='PUT', user=self.user,
+            params={
+                'title': 'modified title'
+            }
+        )
+        self.assertStatus(resp, 403)
+
+        # delete the submission and ensure it is gone
+        resp = self.request(
+            path='/covalic_submission/%s' % id, method='DELETE', user=self.user
+        )
+        self.assertStatusOk(resp)
+        resp = self.request(
+            path='/covalic_submission/%s' % id, user=self.admin
+        )
+        self.assertStatus(resp, 400)
+
+    def testListSubmissions(self):
+        self.generateSubmissionList()
+        resp = self.request(
+            path='/covalic_submission', user=self.user,
+            params={'phaseId': self.phase1['_id']}
+        )
+        self.assertStatusOk(resp)
+        self.assertEqual(len(resp.json), 2)
+
+        resp = self.request(
+            path='/covalic_submission', user=self.user,
+            params={'phaseId': self.phase1['_id'], 'latest': False}
+        )
+        self.assertStatusOk(resp)
+        self.assertEqual(len(resp.json), 6)
+
+        resp = self.request(
+            path='/covalic_submission', user=self.user,
+            params={'phaseId': self.phase1['_id'], 'userId': self.user['_id']}
+        )
+        self.assertStatusOk(resp)
+        self.assertEqual(len(resp.json), 3)
+
+    def testPostScore(self):
+        submission = self.createSubmission(self.phase1, self.user, 'submission')
+
+        score = json.dumps([{
+            'dataset': 'Average',
+            'metrics': [{
+                'name': 'accuracy',
+                'value': 0.5
+            }]
+        }])
+        # users cannot submit scores
+        resp = self.request(
+            path='/covalic_submission/%s/score' % str(submission['_id']),
+            method='POST', user=self.user,
+            body=score, type='application/json'
+        )
+        self.assertStatus(resp, 403)
+
+        # post a score as admin
+        resp = self.request(
+            path='/covalic_submission/%s/score' % str(submission['_id']),
+            method='POST', user=self.admin,
+            body=score, type='application/json'
+        )
+        self.assertStatus(resp, 200)
+
+        # assert the score was added
+        resp = self.request(
+            path='/covalic_submission/%s' % str(submission['_id']), user=self.user
+        )
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.json['overallScore'], 0.5)
```

Comment:
With multiple weighted metrics, we can check that the overallScore is computed correctly. We can also add a check for the computed 'Average' dataset score.